lib/mutant/reporter/hash/printer/mutation_result.rb

```diff
@@ -0,0 +1,149 @@
+module Mutant
+  class Reporter
+    class Hash
+      class Printer
+        # Reporter for mutation results
+        class MutationResult < self
+
+          delegate :mutation, :test_result
+
+          DIFF_ERROR_MESSAGE =
+            'BUG: Mutation NOT resulted in exactly one diff hunk. Please report a reproduction!'.freeze
+
+          MAP = {
+            Mutant::Mutation::Evil    => :evil_details,
+            Mutant::Mutation::Neutral => :neutral_details,
+            Mutant::Mutation::Noop    => :noop_details
+          }.freeze
+
+          NEUTRAL_MESSAGE =
+            "--- Neutral failure ---\n" \
+            "Original code was inserted unmutated. And the test did NOT PASS.\n" \
+            "Your tests do not pass initially or you found a bug in mutant / unparser.\n" \
+            "Subject AST:\n" \
+            "%s\n" \
+            "Unparsed Source:\n" \
+            "%s\n" \
+            "Test Result:\n".freeze
+
+          NO_DIFF_MESSAGE =
+            "--- Internal failure ---\n" \
+            "BUG: Mutation NOT resulted in exactly one diff hunk. Please report a reproduction!\n" \
+            "Original unparsed source:\n" \
+            "%s\n" \
+            "Original AST:\n" \
+            "%s\n" \
+            "Mutated unparsed source:\n" \
+            "%s\n" \
+            "Mutated AST:\n" \
+            "%s\n".freeze
+
+          NOOP_MESSAGE    =
+            "---- Noop failure -----\n" \
+            "No code was inserted. And the test did NOT PASS.\n" \
+            "This is typically a problem of your specs not passing unmutated.\n" \
+            "Test Result:\n".freeze
+
+          FOOTER = '-----------------------'.freeze
```

Comment:
All the messages here are duplicated, lets push them down to the proposed printer class till we find a better place.